src/ui/Analytics/Analytics.ts

```diff
@@ -158,7 +162,69 @@ export class Analytics extends Component {
      * Default value is `undefined`, and the value of this parameter will fallback to the organization used for the
      * search endpoint.
      */
-    organization: ComponentOptions.buildStringOption()
+    organization: ComponentOptions.buildStringOption(),
+
+    /**
+     * Whether to push Coveo usage analytics events to the Google Tag Manager [data layer](https://developers.google.com/tag-manager/devguide#datalayer).
+     *
+     * **Notes:**
+     * - Set this option to `false` to ensure that the component does _not_ populate the data layer, even if a GTM container snippet is present in the page.
+     * - Set this option to `true` to ensure that the component populates the data layer even if a GTM container snippet cannot be found in the page.
+     *
+     * See also [`gtmDataLayerName`]{@link Analytics.options.gtmDataLayerName}.
+     *
+     * By default, if a standard GTM container snippet is found in the page, the component automatically attempts to push to its initialized data layer (see [Google Tag Manager Quick Start Guide](https://developers.google.com/tag-manager/quickstart)).
+     */
+    pushToGtmDataLayer: ComponentOptions.buildBooleanOption({
+      postProcessing: (value?: boolean, options?: IAnalyticsOptions) => {
+        const logger = new Logger('Analytics');
+        let optionValue = false;
+        if (typeof value === 'boolean') {
+          optionValue = value;
+        } else {
+          const includesGtmScript = new RegExp(`https:\/\/www\.googletagmanager\.com\/gtm\.js`);
+          const initializesGtmScript = new RegExp(`\(window,document,'script','(\\w+)','GTM-[A-Z0-9]{7}'\)`);
+          const gtmContainerSnippet = _.find(document.querySelectorAll('script'), scriptElement => {
+            // Could yield false positives (e.g., if the innerText is commented out).
+            return includesGtmScript.test(scriptElement.innerText) && initializesGtmScript.test(scriptElement.innerText);
+          });
+          if (gtmContainerSnippet) {
+            const match = initializesGtmScript.exec(gtmContainerSnippet.innerText);
+            // Attempt to extract data layer name from GTM container snippet
```

Comment:
It would arguably be cleaner to do this in the `gtmDataLayerName` option post processing, but this would require the regexes to be tested again (worst case, against the inner text of all script tags the page).

Doing this here also means that if `pushToGtmDataLayer` is left undefined and the `gtmDataLayerName` value is explicitly specified in the component configuration, the specified value will always be overridden if a GTM container snippet is found in the page. My intuition is this would typically "fix" a configuration mistake, but it would also make sense if an explicitly specified option value always took precedence over an automatically generated one.

Thoughts?

Reply:
I share your concerns. In trying to help the user, we are adding complexity. I left a comment above suggesting to simplify by not auto-detecting values.